controllers/subscription_controller.go

```diff
@@ -138,10 +140,13 @@ func (r *SubscriptionReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	// create componentplan
+	componentPlanName := corev1alpha1.GenerateComponentPlanName(sub, latestVersionFetch.Version)
 	if err = r.CreateComponentPlan(ctx, sub, latestVersionFetch); err != nil {
 		logger.Error(err, "Failed to create component plan")
+		r.Recorder.Eventf(sub, "CreateComponentPlan", "Fail", "failed to create componentPlan %s with error: %s", componentPlanName, err)
```

Comment:
define event types in `subscription.go`similar to https://github.com/kubebb/core/pull/123/commits/074ee24a026bbc81543e170563b8b5a945d0df0a#diff-cd75f6ee7557f93b1b1c91bec463418a9547915348ab4b846a45f91712012e87R21